### PR TITLE
Add Glonass CA ranging code generation function and unit test

### PR DESCRIPTION
**Pull request by pmiettinen**

### CODE REVIEW

**Comment by pmiettinen** [comment]
To be thorough, the generated code should be checked against a result produced by some other party.

**Comment by ljbade** [comment]
@pmiettinen you could check against https://github.com/ljbade/gnss-prn-gen/blob/master/prngen/prngen/prns.c#L2526-L2530

**Comment by pmiettinen** [comment]
@ljbade thanks, looks like a match, I just packed it to same format a moment ago
static const u8 glo_l1l2ca_code[] = {
  0xFE, 0x0F, 0x7C, 0x5C, 0xC8, 0x25, 0x3B, 0x47, 0x9F, 0x36, 0x2A,
  0x47, 0x1B, 0x57, 0x13, 0x11, 0x00, 0x84, 0x61, 0x39, 0x56, 0x1B,
  0xD3, 0x72, 0x28, 0x56, 0x9F, 0xB2, 0x4B, 0x7E, 0x4D, 0x4C, 0xC0,
  0x63, 0x28, 0xD2, 0xFE, 0x8B, 0x1D, 0x65, 0x9E, 0x3E, 0xE8, 0x35,
  0xB7, 0x60, 0xB5, 0xF5, 0x50, 0x29, 0x5E, 0x5D, 0xC0, 0xE7, 0x49,
  0xEB, 0xA8, 0x90, 0xCE, 0x17, 0xB6, 0x68, 0x77, 0x86
};

File: peregrine/include/generateGLOCAcode.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  """
+  code = '111111111'
+  output = ''
+  while len(output) < 511:
+    output = code[6] + output
+    if int(code[4]) ^ int(code[8]):
+      code = '1' + code
+    else:
+      code = '0' + code
+  return output[::-1]
```
Comment (valeri-atamaniouk) [inline peregrine/include/generateGLOCAcode.py]: Please make a return of the same type as other code generators.

Comment (valeri-atamaniouk) [inline peregrine/include/generateGLOCAcode.py]: The output seems to be inverted.

File: tests/test_generateGLOCAcode.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+from peregrine.include.generateGLOCAcode import generateGLOCAcode
+
+
+def test_generateGLOCAcode():
+  code = '1111111000001111011111000101110011001000001001010011101101000111100111110011011000101010010001110001101101010111000100110001000100000000100001000110000100111001010101100001101111010011011100100010100001010110100111111011001001001011011111100100110101001100110000000110001100101000110100101111111010001011000111010110010110011110001111101110100000110101101101110110000010110101111101010101000000101001010111100101110111000000111001110100100111101011101010001001000011001110000101111011011001101000011101111000011'
```
Comment (valeri-atamaniouk) [inline tests/test_generateGLOCAcode.py]: 79 characters limit.

File: peregrine/include/generateGLOCAcode.py (hunk 1 above)
Comment (valeri-atamaniouk) [inline peregrine/include/generateGLOCAcode.py]: Add a module variable to access the code instead of calling the function every time.

**Comment by ljbade** [comment]
@valeri-atamaniouk Why do you think the output is inverted?

**Comment by valeri-atamaniouk** [comment]
@ljbade @pmiettinen If you take a formula 1+x^5+x^9, and the initial register is 0b111111111, the first bit can't be 1. It is 0.

**Comment by valeri-atamaniouk** [comment]
@ljbade @pmiettinen I was wrong.

**Comment by pmiettinen** [comment]
@valeri-atamaniouk Your comments are now addressed. Please review.

File: peregrine/include/glo_ca_code.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    else:
+      output[i] = -1
+    if int(code[4]) ^ int(code[8]):
+      code = '1' + code[:8]
+    else:
+      code = '0' + code[:8]
+  return output
+
+
+def readonly(value):
```
Comment (cbeighley) [inline peregrine/include/glo_ca_code.py]: Could we make this more similar to the other code generation modules? See the following:
- https://github.com/adel-mamin/peregrine/blob/peregrine-l2c-update/peregrine/include/generateL2CMcode.py
- https://github.com/swift-nav/peregrine/blob/master/peregrine/include/generateCAcode.py

Steps should be:
- Remove `readonly`, `glo_ca_code`, and `tmp`
- Add `GLOCode = generate_glo_ca_code` to the bottom of the module
- Rename file to `generateGLOcode.py`

File: tests/test_glo_ca_code.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+from peregrine.include.glo_ca_code import value
+
+import numpy as np
+
+
+def test_glo_ca_code():
```
Comment (cbeighley) [inline tests/test_glo_ca_code.py]: Looks good. Please update name to `tests/test_generateGLOcode.py'.

**Comment by pmiettinen** [comment]
@cbeighley Your comments are addressed. Please review.

**Comment by pmiettinen** [comment]
@cbeighley Please review.

**Comment by pmiettinen** [comment]
Superseded by https://github.com/swift-nav/peregrine/pull/43

**Pull request closed without merging**